lib/modules/manager/npm/extract/common/dependency.ts

```diff
@@ -105,7 +104,8 @@ export function extractDependency(
       dep.packageName = valSplit[0] + '@' + valSplit[1];
       dep.currentValue = valSplit[2];
     } else {
-      logger.debug('Invalid npm package alias: ' + dep.currentValue);
+      dep.packageName = depName;
```

Comment:
you need another condition and check valSplit===1 and only then do this, otherwise log debug message as now.

Reply:
@viceice Thank you for your comment. I've addressed this and added the check in a logical order and slightly improved the debug logging as:
```
// from
Invalid npm package alias: npm:^1.x.x
//to
Invalid npm package alias for dependency: lodash:npm:^2.2.7
```

Reply:
Noticed that the code coverage decreased since the else statement will never be executed.
```
// Length of valSplit will always be at least 1
const valSplit = dep.currentValue.replace('npm:', '').split('@');
/**
* so now that there's an explicit condition for valSplit.length === 1
* the else statement will never execute
*/
```

Reply:
Or you could add a test with a split of 4

Reply:
I could, but I under how realistic that is in a real world scenario. Please see https://github.com/renovatebot/renovate/pull/24568#discussion_r1334239973